cmd/object-handlers_test.go

```diff
@@ -527,11 +540,28 @@ func testAPIGetObjectHandler(obj ObjectLayer, instanceType, bucketName string, a
 		// read the response body.
 		actualContent, err := ioutil.ReadAll(rec.Body)
 		if err != nil {
-			t.Fatalf("Test %d: %s: Failed parsing response body: <ERROR> %v", i+1, instanceType, err)
+			t.Fatalf("Test %d: %s: Failed reading response body: <ERROR> %v", i+1, instanceType, err)
 		}
+
+		if rec.Code == http.StatusOK || rec.Code == http.StatusPartialContent {
+			if !bytes.Equal(testCase.expectedContent, actualContent) {
+				t.Errorf("Test %d: %s: Object content differs from expected value %s, got %s", i+1, instanceType, testCase.expectedContent, string(actualContent))
+			}
+			continue
+		}
+
 		// Verify whether the bucket obtained object is same as the one created.
-		if !bytes.Equal(testCase.expectedContent, actualContent) {
-			t.Errorf("Test %d: %s: Object content differs from expected value %s, got %s", i+1, instanceType, testCase.expectedContent, string(actualContent))
+		actualError := &APIErrorResponse{}
+		if err = xml.Unmarshal(actualContent, actualError); err != nil {
```

Comment:
The following code is repeated in several places (6) :
```
actualError := &APIErrorResponse{}
if err = xml.Unmarshal(actualContent, actualError); err != nil {
	t.Fatalf("Test %d: %s: Failed parsing response body: <ERROR> %v", i+1, instanceType, err)
}

if actualError.BucketName != testCase.bucketName {
	t.Fatalf("Test %d: %s: Unexpected bucket name, expected %s, got %s", i+1, instanceType,testCase.bucketName, actualError.BucketName)
}

if actualError.Key != testCase.objectName {
	t.Fatalf("Test %d: %s: Unexpected object name, expected %s, got %s", i+1, instanceType,testCase.objectName, actualError.Key)
}
```
Even the following code can be included although it differs a little bit
```
if rec.Code == http.StatusOK || rec.Code == http.StatusPartialContent {
	if !bytes.Equal(testCase.expectedContent, actualContent) {
		t.Errorf("Test %d: %s: Object content differs from expected value %s, got %s", i+1, instanceType,testCase.expectedContent, string(actualContent))
	}
	continue
}
```
Can you DRY it?
Here are the file name and line numbers where this code is repeated:

FileName | Line#s
--------------|---------
cmd/object-handlers_test.go | 554 ~ 565
   //                                         | 603 ~ 614
   //                                         | 2847 ~ 2858
cmd/test-utils_test.go
   //                                         | 1785 ~ 1796
   //                                         | 1846 ~ 1857
   //                                         | 1901 ~ 1912

Reply:
These tests will be removed @ebozduman in a separate run - this PR is simply to make that they don't fail.

Reply:
So for now this PR is primarily to make our error XML dynamic..

Reply:
OK